vslm/global_object_manager.go

```diff
@@ -0,0 +1,659 @@
+		This:      this.Reference(),
+		Datastore: datastore.Reference(),
+	}
+
+	_, err := methods.VslmScheduleReconcileDatastoreInventory(ctx, this.c, req)
+
+	return err
+}
+
+func (this *GlobalObjectManager) CreateSnapshot(ctx context.Context, id vim.ID, description string) (*Task, error) {
+	req := types.VslmCreateSnapshot_Task{
+		This:        this.Reference(),
+		Id:          id,
+		Description: description,
+	}
+
+	res, err := methods.VslmCreateSnapshot_Task(ctx, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewTask(this.c, res.Returnval), nil
+}
+
+func (this *GlobalObjectManager) DeleteSnapshot(ctx context.Context, id vim.ID, snapshotID vim.ID) (*Task, error) {
+	req := types.VslmDeleteSnapshot_Task{
+		This:       this.Reference(),
+		Id:         id,
+		SnapshotId: snapshotID,
+	}
+
+	res, err := methods.VslmDeleteSnapshot_Task(ctx, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewTask(this.c, res.Returnval), nil
+}
+
+func (this *GlobalObjectManager) RetrieveSnapshotInfo(ctx context.Context, id vim.ID) (
+	[]vim.VStorageObjectSnapshotInfoVStorageObjectSnapshot, error) {
+	req := types.VslmRetrieveSnapshotInfo{
+		This: this.Reference(),
+		Id:   id,
+	}
+
+	res, err := methods.VslmRetrieveSnapshotInfo(ctx, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return res.Returnval.Snapshots, nil
+}
+
+func (this *GlobalObjectManager) CreateDiskFromSnapshot(ctx context.Context, id vim.ID, snapshotId vim.ID, name string,
+	profile []vim.VirtualMachineProfileSpec, crypto *vim.CryptoSpec,
+	path string) (*Task, error) {
+	req := types.VslmCreateDiskFromSnapshot_Task{
+		This:       this.Reference(),
+		Id:         id,
+		SnapshotId: snapshotId,
+		Name:       name,
+		Profile:    profile,
+		Crypto:     crypto,
+		Path:       path,
+	}
+
+	res, err := methods.VslmCreateDiskFromSnapshot_Task(ctx, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewTask(this.c, res.Returnval), nil
+}
+
+func (this *GlobalObjectManager) Revert(ctx context.Context, id vim.ID, snapshotID vim.ID) (*Task, error) {
+	req := types.VslmRevertVStorageObject_Task{
+		This:       this.Reference(),
+		Id:         id,
+		SnapshotId: snapshotID,
+	}
+
+	res, err := methods.VslmRevertVStorageObject_Task(ctx, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewTask(this.c, res.Returnval), nil
+}
+
+func (this *GlobalObjectManager) QueryChangedDiskAreas(ctx context.Context, id vim.ID, snapshotId vim.ID, startOffset int64,
+	changeId string) (*vim.DiskChangeInfo, error) {
+	req := types.VslmQueryChangedDiskAreas{
+		This:        this.Reference(),
+		Id:          id,
+		SnapshotId:  snapshotId,
+		StartOffset: startOffset,
+	}
+
+	res, err := methods.VslmQueryChangedDiskAreas(ctx, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return &res.Returnval, nil
+}
+
+func (this *GlobalObjectManager) queryGlobalCatalogSyncStatus(ct context.Context) ([]types.VslmDatastoreSyncStatus, error) {
```

Comment:
Are these 2 methods intentionally not exported or should that be `Q`?

Reply:
Forgot those

Reply:
Should be fixed now